feat(swan_r5): adds a basic STM32 RTC implementation.

m-mcgowan opened the pull request:
This adds a simple STM32 implementation of the Real Time Clock. 

## Testing

This test script outputs the current time every second, via both the RTC and `time.time()`. 

```
import time
import rtc

r = rtc.RTC()
r.datetime = time.struct_time((2022, 6, 13, 3, 4, 5, -1, -1, -1))

while True:
    print(r.datetime)
    time.sleep(1)
    print(time.time())

tannewt commented:
Looks like the build failed because I moved `translate.h` on you. It is now `#include "supervisor/shared/translate/translate.h"`. Would you mind turning RTC on for all STM builds it will work with?

m-mcgowan commented:
> Looks like the build failed because I moved `translate.h` on you. It is now `#include "supervisor/shared/translate/translate.h"`. Would you mind turning RTC on for all STM builds it will work with?

I've fixed up the include locally, and swan_r5 builds.  But I'm seeing build errors on other stm boards after pulling in `main`

```
In file included from ../../py/obj.h:33,
                 from ../../shared-module/displayio/Palette.h:33,
                 from ../../shared-bindings/displayio/Palette.h:30,
                 from build-stm32f4_discovery/autogen_display_resources-en_US.c:2:
../../py/qstr.h:42:14: fatal error: genhdr/qstrdefs.enum.h: No such file or directory
     #include "genhdr/qstrdefs.enum.h"
              ^~~~~~~~~~~~~~~~~~~~~~~~
 ```
 
 This is with a clean build. Any thoughts what might be the cause here?
 
 EDIT: It looks like it has something to do with parallel builds and the `-j` option to make.

tannewt commented:
> This is with a clean build. Any thoughts what might be the cause here?

I think we're missing a dependency in our Makefile and the -j leads to a race condition. It's not an issue with your code. (It's a change I made.)

tannewt reviewed (approved):
Thank you!
